internal/server/data/destination.go

```diff
@@ -176,11 +176,25 @@ func ListDestinations(tx ReadTxn, opts ListDestinationsOptions) ([]models.Destin
 }
 
 func DeleteDestination(tx WriteTxn, id uid.ID) error {
+	// Delete any grants for the destination before deleting it
 	stmt := `
+		UPDATE grants SET deleted_at = ?
+		WHERE resource =
+			(SELECT name FROM destinations
+			WHERE id = ? AND organization_id = ? AND deleted_at is null)
```

Comment:
This will match `ssh` destinations fine, but only makes `kubernetes` grants for the cluster, right?  Kubernetes grants for specific namspaces have a resource of `<destinationName>.<namespace>`. 

`data.ListGrants` uses `data.grantsByDestination` to perform that query. You might be able to use that here as well.  I think if you move this new delete after the `UPDATE destinations`, you can check the result from `tx.Exec` to determine if the destination was deleted or not. That should avoid the need to `WHERE organization_id = ? AND deleted_at is null` since it's already done by the `UPDATE destinations`.

Both queries are performed in the same transaction, so I'm guessing order doesn't matter (other than switching them might make it easier to do the cleanup of grants, and matches how we do this in `DeleteProviders` and `DeleteIdentities`).

Reply:
This is a great catch. This was a little bit tricky to sort out, due to using the destination name inside of a grant instead of the IDs. I've changed this to use the query builder, and to use the `grantsByDestination()` function, which I think is pretty straight forward.

I did not drop the `WHERE organization_id = ?` guard though, because a different org could have a destination named the same thing. I did explore using the IDs of each of the grants which would have enabled me to drop the guard, but that ended up being a lot more complicated.